feat: provide `--home-network` arg for `add` cmd

jacderida opened the pull request:
The `add` command now enables the user to pass the `--home-network` argument to the `safenode` service.

BREAKING CHANGE: the addition of the `home_network` field to `NodeServiceData` will mean this version of the node manager will not be able to deserialize old node registry files.

## Description

reviewpad:summary

joshuef commented on sn_node/src/node/node_starter.rs:
```diff
@@ -141,7 +141,7 @@ async fn bootstrap_node(
     join_timeout: Duration,
 ) -> Result<(Arc<RwLock<MyNode>>, CmdChannel, EventReceiver)> {
-    let (connection_event_tx, mut connection_event_rx) = mpsc::channel(100);
+    let (connection_event_tx, mut connection_event_rx) = mpsc::channel(10_00);
```
i was intending 10_000 :D so yeh, it's wrong in all fashions!

jacderida committed:
feat: provide `--home-network` arg for `add` cmd

The `add` command now enables the user to pass the `--home-network` argument to the `safenode`
service.

BREAKING CHANGE: the addition of the `home_network` field to `NodeServiceData` will mean this
version of the node manager will not be able to deserialize old node registry files.
